trl/models/utils.py

```diff
@@ -132,6 +137,69 @@ def setup_chat_format(
     return model, tokenizer
 
 
+def setup_chat_template(
+    model: PreTrainedModel,
+    tokenizer: PreTrainedTokenizer,
+    source: str,
+    resize_to_multiple_of: Optional[int] = 64,
+) -> tuple[PreTrainedModel, PreTrainedTokenizer]:
+    """
+    Sets up the chat template for the tokenizer and model, copying the chat template from a source tokenizer.
+
+    This function:
+    - Get the chat template from a pretrained tokenizer specified by `source` and sets it for the provided tokenizer.
+    - Update the tokenizer's EOS token and add it to the special tokens.
+    - Resize the model's token embeddings to accommodate the new special tokens.
+    - Update the model's configuration and generation configuration to use the new EOS token.
+
+    Args:
+        model (`PreTrainedModel`):
+            Model to update.
+        tokenizer (`PreTrainedTokenizer`):
+            Tokenizer to update with a new chat template.
+        source (`str`):
+            Name or path of a pretrained tokenizer from which to copy the chat template.
+        resize_to_multiple_of (`int` or `None`, *optional*, defaults to `64`):
+            Resize the model's token embeddings to a multiple of this value. If `None`, no resizing is performed.
+
+    Returns:
+        model (`PreTrainedModel`):
+            The updated model with resized token embeddings and EOS token configured.
+        tokenizer (`~transformers.PreTrainedTokenizer`):
+            The updated tokenizer with the chat template and special tokens applied.
+
+    Example:
+    ```python
+    from transformers import AutoModelForCausalLM, AutoTokenizer
+    from trl import setup_chat_template
+    model = AutoModelForCausalLM.from_pretrained("meta-llama/Llama-3.2-1B")
+    tokenizer = AutoTokenizer.from_pretrained("meta-llama/Llama-3.2-1B")
+    model, tokenizer = setup_chat_template(model, tokenizer, source="Qwen/Qwen3-0.6B")
+    ```
+    """
+    # Load the source tokenizer
+    tokenizer_source = AutoTokenizer.from_pretrained(source)
+
+    # Set the chat template for the tokenizer
+    tokenizer.chat_template = tokenizer_source.get_chat_template()
+
+    # Set the special tokens for the tokenizer
+    tokenizer.add_special_tokens({"additional_special_tokens": [tokenizer_source.eos_token]})
```

Comment:
What if the source tokenizer's chat template contains other special tokens such as `<think>`, <|im_start|> etc , would we not need to add these to our tokenizer? Or should this be done by the user?

Reply:
Done in f1fbe73d647d04161ba93c12c66ea6057947a0bc